src/main/java/org/springframework/data/redis/connection/lettuce/LettuceConverters.java

```diff
@@ -512,7 +513,13 @@ public static RedisURI sentinelConfigurationToRedisURI(RedisSentinelConfiguratio
 
 			RedisURI.Builder sentinelBuilder = RedisURI.Builder.redis(sentinel.getHost(), sentinel.getPort());
 
-			sentinelPassword.toOptional().ifPresent(sentinelBuilder::withPassword);
+			String sentinelUsername = sentinelConfiguration.getSentinelUsername();
+			if (StringUtils.hasText(sentinelUsername)) {
```

Comment:
We should set the authentication with username only if a password is present in addition to the username check. Having a username without a password isn't supported in the authentication scheme.